### PR TITLE
Show All Exams Checkbox Added

### PR DESCRIPTION
**Pull request by JayasriRamu**
Added One Checkbox. Main Idea taken from Ramu

### CODE REVIEW

File: src/app/model/data.service.ts [hunk 1]
```diff
@@ -98,6 +100,7 @@ export class DataService {
   public activeUser: string
   public disableHotkeys = false
   public titleFilter = ''
+  public isCheck = false
```
Comment (svramu) [inline src/app/model/data.service.ts]: This could be named as 'showAll'.

File: src/app/model/data.service.ts [hunk 2]
```diff
@@ -149,7 +167,7 @@ export class DataService {
 
   public listResults(eid: string): ExamResult[] {
     return this.results
-      .filter(r => r.exam.id === eid)
+      .filter(r => (r.exam.id === eid))
```
Comment (svramu) [inline src/app/model/data.service.ts]: No change required.

File: src/app/model/data.service.ts [hunk 3]
```diff
@@ -290,9 +308,23 @@ export class DataService {
     return eid + '.pending'
   }
 
+  public doneId(eid: string) {
```
Comment (svramu) [inline src/app/model/data.service.ts]: Method not required.

File: src/app/model/data.service.ts [hunk 4]
```diff
@@ -290,9 +308,23 @@ export class DataService {
...
+    return eid + '.done'
+  }
   private shadowPendingExam(eid: string) {
     let exam = this.cache[eid]
-    let rid = this.pendingId(eid)
+    // let rid = this.pendingId(eid)
```
Comment (svramu) [inline src/app/model/data.service.ts]: This change section is not required.

File: src/app/student-dash/student-dash.component.html [hunk 5]
```diff
@@ -19,8 +23,7 @@
       </mat-form-field>
     </div>
     <div *ngIf="service.isAdmin" fxFlexAlign="baseline">
-      <button mat-raised-button color="primary" 
-          id="create-exam-button" (click)="createExam()">Create Exam!</button>
+      <button mat-raised-button color="primary" id="create-exam-button" (click)="createExam()">Create Exam!</button>
```
Comment (svramu) [inline src/app/student-dash/student-dash.component.html]: Change not required. Let us not change the formatting unless very essential, please.

File: src/app/student-dash/student-dash.component.html [hunk 6]
```diff
@@ -29,13 +32,21 @@
   </div>
 
   <br>
-  <div class="center">
-    <mat-card *ngFor="let ex of service.filteredExams() | paginate: { id:'exams', itemsPerPage: 10, currentPage: pageExam }"
-        [class.pending]="ex.isPending()"
-        style="margin-bottom: 4px;"
-        >
+
+  <!--  <div class="center" [style.visibility]="service.isCheck ? 'visible' : 'hidden'" id="isCheck">
```
Comment (svramu) [inline src/app/student-dash/student-dash.component.html]: Please, remove all commented blocks.

File: src/app/student-dash/student-dash.component.html [hunk 7]
```diff
@@ -69,13 +115,8 @@
...
   <br>
-  <a
-    style="padding: 4px; float: right; text-align: right;"
-    (mouseenter)="toggleOK()"
-    (mouseleave)="toggleOK()"
-    [@testAnim]="ok"
-    href="https://github.com/Tattvum/edutvum-exam"
-    target="_blank"
-    >https://github.com/Tattvum/edutvum-exam</a>
+  <a style="padding: 4px; float: right; text-align: right;" (mouseenter)="toggleOK()" (mouseleave)="toggleOK()"
```
Comment (svramu) [inline src/app/student-dash/student-dash.component.html]: Please avoid dispensable formatting. This is to aid clarity of changes required for the feature.

File: src/app/model/data.service.ts [hunk 8]
```diff
@@ -84,11 +84,13 @@ interface Cache {
   [id: string]: Exam
 }
 
+
 @Injectable()
 export class DataService {
   private userCache: UserCache = {}
   private cache: Cache = {}
   private pendingResult: ExamResult
+  private doneResult: ExamResult
```
[the remark this answers is not in the document]
Comment (JayasriRamu) [reply src/app/model/data.service.ts]: Agreed. Done.

File: src/app/model/data.service.ts (hunk 1 above)
Comment (JayasriRamu) [reply src/app/model/data.service.ts]: Agreed. Done.

File: src/app/model/data.service.ts (hunk 2 above)
Comment (JayasriRamu) [reply src/app/model/data.service.ts]: Agreed. Done.

File: src/app/model/data.service.ts (hunk 3 above)
Comment (JayasriRamu) [reply src/app/model/data.service.ts]: Agreed. Done.

**Comment by JayasriRamu** [comment]
All review comments done.

**Pull request merged**